### PR TITLE
Update Node version to >= 12

### PR DESCRIPTION
**Pull request by safaiyeh**
## Summary
Update engines to node >= 12

## Changelog

[General] [Changed] - Update node engine to version 12

### CODE REVIEW

**Comment by safaiyeh** [comment]
cc @hramos

**Review by cpojer — APPROVED** [review]
👍

**Comment by cpojer** [comment]
It looks like at least some CI steps are still using node 10. Could you update those?

**Comment by safaiyeh** [comment]
> It looks like at least some CI steps are still using node 10. Could you update those?

Sounds good, I'll check it out tomorrow.

**Comment by safaiyeh** [comment]
Had to update to the latest `docker-android` to support Node >= 12, which also requires updating to Android SDK 21.

There is currently an open issue here: https://github.com/react-native-community/docker-android/issues/93 concerning the AVD which is breaking test_android

**Comment by safaiyeh** [comment]
test_android also seems to be missing JAVA_HOME as well

**Review by hramos — APPROVED** [review]
Thanks for working on this!

**Review by gengjiawen — APPROVED** [review]
Looks like `.nvmrc` can also be removed. LGTM.

File: scripts/.tests.env [hunk 1]
```diff
@@ -10,7 +10,7 @@ export ANDROID_SDK_BUILD_API_LEVEL="28"
 # Google APIs for Android level
 export ANDROID_GOOGLE_API_LEVEL="23"
 # Minimum Android API Level we target
-export ANDROID_SDK_TARGET_API_LEVEL="19"
+export ANDROID_SDK_TARGET_API_LEVEL="21"
```
Comment (gengjiawen) [inline scripts/.tests.env:13]: This can be set to 29. Google is forcing it https://developer.android.com/distribute/best-practices/develop/target-sdk.

**Comment by gengjiawen** [comment]
You missed `.circleci/Dockerfiles/Dockerfile.android`.

File: scripts/.tests.env (hunk 1 above)
Comment (fkgozali) [reply scripts/.tests.env:13]: From ReactAndroid/build.gradle:

```
        minSdkVersion(21)
        targetSdkVersion(28)
```

So if the comment is right, then 21 is correct, unless the comment is wrong:

> # Minimum Android API Level we target

Comment (gengjiawen) [reply scripts/.tests.env:13]: Sorry, judge from the name, I think it's targetSdkVersion.

**Comment by safaiyeh** [comment]
Made

> You missed `.circleci/Dockerfiles/Dockerfile.android`.

Appreciate it! updated that

**Comment by fkgozali** [comment]
Hmm, the test_android/test_docker are still failing, are we missing something?

**Comment by gengjiawen** [comment]
> Hmm, the test_android/test_docker are still failing, are we missing something?

Looks buck config get wrong
```console
Step 14/24 : RUN buck fetch ReactAndroid/src/test/java/com/facebook/react/modules
 ---> Running in 3ea762ab4fe4
Not using buckd because watchman isn't installed.
Picked up JAVA_TOOL_OPTIONS: -Dfile.encoding=UTF8
PARSING BUCK FILES: FINISHED IN 4.3s
No build file at Libraries/BUCK when resolving target //Libraries:generated_java_modules-FBReactNativeSpec.
```

**Pull request closed without merging**